ramips: add Asus RP-AC56 support

MacArrow opened the pull request:
This change adds support for Asus RP-AC56 which
is a dual band dual core 802.11ac access point

Specifications:
	CPU:   MT7621A dual-core 880MHz
	RAM:   64MB DDR2
	FLASH: 16MB MX25L12805D NOR SPI
	WIFI:  2.4GHz 2x2 MT7603 b/g/n PCI
	WIFI:  5GHz 2x2 MT7662 a/b/ac PCI
	ETH:   1xLAN 1000base-T integrated
	LED:   Power, 2.4GHz WIFI, 5GHz WIFI
	BTN:   WPS, Reset
	UART:  Near ETH port, from ETH: 3V3-TxD-GND-RxD 57600 8n1
	MISC:  Audio support

Installation:
	1. Update from factory image
	   Plese use *factory.trx image to update
	2. Update using recovery mode
	   - while holdig "reset" button power on device
	   - keep holding "reset" until power led is flashing yellow
	   - push firmware image (can be factory trx or openwrt sysupgrade)
	     using tftp client in binary mode to 192.168.1.1

Notes:
	This board has only two MAC addresses programmed in "factory" partition:
	- MAC for WLAN0 (2.4GHz) at offset 0x0004
	- MAC for WLAN1 (5GHz) at offset 0x8004
	- stock firmware re-uses WLAN0 MAC for Ethernet so I am doing it too
	- no valid addresses found in 0x28, 0x2e, 0xe000 and 0xe006

MacArrow committed:
firmware-utils: mkrtn56uimg: rename and accept product ID from cmd args

This change renames mkrtn56uimg to mkasusfw and adds an ability
to take a product ID string from a command line argument.
Updating other boards using this tool

Signed-off-by: Vitaly Makarov <vit.macarrow@gmail.com>

MacArrow committed:
kernel: mtdsplit: add Asus extended uImage support

In some of its products Asus uses an extended uImage format
where ih_name field from original uImage header is prepended
with kernel and rootfs versions and appended with ih_ksz.
In such header, ih_size field contains a size of the whole
image and kernel size is stored in ih_ksz. This means that
in order to calculate an offset to rootfs, ih_ksz should
be used instead of ih_size.
This change adds a new compartible target for partition
"asus,uimage" and a logic to detect Asus extended header
and use a proper rootfs offset from it.

Signed-off-by: Vitaly Makarov <vit.macarrow@gmail.com>

MacArrow committed:
ramips: adding support for Asus RP-AC56

Specifications:
	CPU:     MT7621A dual-core 880MHz
	RAM:     64MB DDR2
	FLASH:   16MB MX25L12805D NOR SPI
	WIFI:    2.4GHz 2x2 MT7603 b/g/n PCI
	WIFI:    5GHz 2x2 MT7662 a/b/ac PCI
	ETH:     1xLAN 1000base-T integrated
	SWITCH:  MT7530 Port 0: LAN, Port 6: CPU
	LED:     Power, 2.4GHz WIFI, 5GHz WIFI
	BTN:     WPS, Reset
	UART:    Near ETH port, from ETH: 3V3-TxD-GND-RxD 57600 8n1
	MISC:    Audio support

Installation:
	1. Update from factory image
	   Plese use *factory.trx image to update
	2. Update using recovery mode
	   - while holdig "reset" button power on device
	   - keep holding "reset" until power led is flashing yellow
	   - push firmware image (can be factory trx or openwrt sysupgrade)
	     using tftp client in binary mode to 192.168.1.1

Notes:
	This board has only two MAC addresses programmed in "factory" partition:
	- MAC for WLAN0 (2.4GHz) at offset 0x0004
	- MAC for WLAN1 (5GHz) at offset 0x8004
	- stock firmware re-uses WLAN0 MAC for Ethernet so I am doing it too
	- no valid addresses found in 0x28, 0x2e, 0xe000 and 0xe006

Signed-off-by: Vitaly Makarov <vit.macarrow@gmail.com>

adrianschmutzler commented on tools/firmware-utils/src/mkrtn56uimg.c:
```diff
@@ -252,12 +253,12 @@ int
 {
 	int 		opt;
-	char 		*filename, *progname;
+	char 		*filename, *progname, *productid = "RT-N56U";
```
I'd prefer to remove the (now arbitrary) default value here entirely and add it as parameter to the existing device(s) in image/*.mk as you do it for your new device.

MacArrow replied:
Done

adrianschmutzler commented on target/linux/generic/files/drivers/mtd/mtdsplit/mtdsplit_uimage.c:
```diff
@@ -262,6 +317,33 @@ static struct mtd_part_parser uimage_generic_parser = {
+				      uimage_verify_default);
+}
+
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 9, 0)
```
OpenWrt currently supports 4.14/4.19/5.4, so this "if" is not necessary.

MacArrow replied:
It is for consistency: all or nothing and I did not dare to remove all kernel version checks from that file without having an ability to test all use cases involved

adrianschmutzler replied:
Keep this open, maybe I'll do it separately after you patch has been merged, so it may serve as a reminder to me.

adrianschmutzler replied:
I've dropped the kernel version switches here:
https://github.com/openwrt/openwrt/commit/8ff813e5f45169d0b97f406643197a62ff031b8d

So you can drop them now as well. :-)

MacArrow replied:
Done

adrianschmutzler commented:
Please add "Specifications:" and "Installation:" headlines (or similar) to your device support commit description. Despite, one might make the Installation part more readable by giving it some more structure or converting it into a list.

adrianschmutzler commented on target/linux/ramips/dts/mt7621_asus_rp-ac56.dts:
```diff
@@ -0,0 +1,231 @@
+/dts-v1/;
```
Please add a license (SPDX identifier) like in other DTS files.

adrianschmutzler commented on target/linux/ramips/dts/mt7621_asus_rp-ac56.dts:
```diff
@@ -0,0 +1,231 @@
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <20>;
```
Please try with "gpio-keys" compatible.

adrianschmutzler commented on target/linux/ramips/dts/mt7621_asus_rp-ac56.dts:
```diff
@@ -0,0 +1,231 @@
+&spi0 {
+	status = "okay";
+
+	m25p80@0 {
```
Use more generic "flash@0"

adrianschmutzler commented on target/linux/ramips/dts/mt7621_asus_rp-ac56.dts:
```diff
@@ -0,0 +1,231 @@
+	m25p80@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
```
Please check if higher frequency can be used.

adrianschmutzler commented on target/linux/ramips/dts/mt7621_asus_rp-ac56.dts:
```diff
@@ -0,0 +1,231 @@
+			ralink,function = "gpio";
+		};
+	};
+	i2s_pins: i2s {
```
Add empty line between nodes.

adrianschmutzler commented on target/linux/ramips/dts/mt7621_asus_rp-ac56.dts:
```diff
@@ -0,0 +1,231 @@
+};
+
+&ethernet {
+	mtd-mac-address = <&factory 0x4>;
```
Have you checked whether there are valid addresses in 0x28/0x2e/0xe000/0xe006?

Do the MAC addresses match the assignment in stock firmware?

Further reading: https://openwrt.org/docs/guide-developer/mac.address

MacArrow replied:
The original FW is reusing MAC from wlan0 for eth0 so I am doing the same (wlan0 MAC is @0x0004 and wlan1 MAC is @0x8004)

adrianschmutzler replied:
Still, the same MAC address could be in one of the mentioned locations, which are the recommended ones. Please check.

MacArrow replied:
Sorry, did not mentioned in my original comment: obviously I have  checked it: "factory" partition has only two MACs in it, like I've mentioned before, @0x0004 offset and @0x8004 offset. 
root@mcrpac1:~# cat /proc/mtd
dev:    size   erasesize  name
mtd0: 00030000 00010000 "u-boot"
mtd1: 00010000 00010000 "u-boot-env"
**mtd2: 00010000 00010000 "factory"**

root@mcrpac1:~# hexdump -C -s 0x28 /dev/mtd2
00000028  fd 9b ff ff ff ff ff ff  ff ff ff ff 22 34 0c 20  |............"4. |
root@mcrpac1:~# hexdump -C -s 0x2e /dev/mtd2
0000002e  ff ff ff ff ff ff 22 34  0c 20 ff ff 00 01 00 00  |......"4. ......|
root@mcrpac1:~# hexdump -C -s 0xe000 /dev/mtd2
0000e000  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
root@mcrpac1:~# hexdump -C -s 0xe006 /dev/mtd2
0000e006  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|

adrianschmutzler replied:
Okay. Please preserve that information in the commit message.

MacArrow replied:
Done

adrianschmutzler replied:
Thanks, two typos:

"This board has only two MAC addresses programed in "factory" parition:"
-> "programmed", "partition"

I'd add a fourth point mentioning:
- No valid addresses found in 0x28, 0x2e, 0xe000 and 0xe006

adrianschmutzler commented on target/linux/ramips/dts/mt7621_asus_rp-ac56.dts:
```diff
@@ -0,0 +1,231 @@
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2s_pins>;
+};
+
```
Remove empty line at the end.

adrianschmutzler commented on target/linux/ramips/image/mt7621.mk:
```diff
@@ -146,6 +150,17 @@ define Device/asiarf_ap7621-nv1
+  DEVICE_MODEL := RP-AC56
+  IMAGE_SIZE := 16000k
+  BLOCKSIZE := 64k
+  DEVICE_PACKAGES := kmod-mt7603 kmod-mt76x2 wpad-basic kmod-sound-mt7620 kmod-sound-core kmod-i2c-ralink
```
Please wrap lines after 80 chars (same below).

adrianschmutzler commented on target/linux/ramips/image/mt7621.mk:
```diff
@@ -146,6 +150,17 @@ define Device/asiarf_ap7621-nv1
+  BLOCKSIZE := 64k
+  DEVICE_PACKAGES := kmod-mt7603 kmod-mt76x2 wpad-basic kmod-sound-mt7620 kmod-sound-core kmod-i2c-ralink
+  IMAGES += factory.trx
+  IMAGE/factory.trx := append-kernel | append-rootfs | pad-rootfs | append-metadata | asus-rp-factory -p RP-AC56 -f | check-size $$$$(IMAGE_SIZE)
```
append-metadata should be only required for sysupgrade.

adrianschmutzler commented on target/linux/ramips/mt7621/base-files/lib/preinit/04_handle_checksumming:
```diff
@@ -0,0 +1,33 @@
+#!/bin/sh
```
I haven't checked explicitly, but I'd expect that preinit files (in /lib folder) are sourced and thus the shebang should be removed.

adrianschmutzler commented on target/linux/ramips/mt7621/base-files/etc/board.d/01_leds:
```diff
@@ -17,6 +17,10 @@ led_wlan="$(get_dt_led wlan)"
 case $board in
+asus,rp-ac56)
+	ucidef_set_led_netdev "wlan2g" "WiFi 2.4GHz" "$boardname:green:wifi2g" "wlan0"
+	ucidef_set_led_netdev "wlan5g" "WiFi 5GHz" "$boardname:green:wifi5g" "wlan1"
```
Please use phyXtpt trigger in DTS for WiFi LEDs.

adrianschmutzler commented on target/linux/ramips/mt7621/base-files/lib/preinit/04_handle_checksumming:
```diff
@@ -0,0 +1,33 @@
+#!/bin/sh
+
+do_checksumming_disable() {
+	. /lib/functions.sh
```
This should be unnecessary as well, as it's sourced in the file where preinit files are sourced:
https://github.com/openwrt/openwrt/blob/master/package/base-files/files/etc/preinit

adrianschmutzler commented:
You need to adjust this patch for kernel 5.4, which is now default on mt7621. This will mostly affect network configuration, have a look at this particular patch for inspiration:
https://github.com/openwrt/openwrt/commit/30644bc579e771e4b6dcc165ca966124e0259e1e

MacArrow commented:
Sure, will do. Is 5.4 enabled by default on a tip?

adrianschmutzler commented:
> Sure, will do. Is 5.4 enabled by default on a tip?

What do you mean by "on a tip"? Are you referring to the latest commit in master branch (i.e. what the "tip" would be in Mercurial?). Then the answer is "yes", if you rebase on the latest master, then m7621 will have kernel 5.4 by default (but only this _subtarget_, all other subtargets are still on 4.14, ...)

MacArrow commented:
Yes, got it, thank you, will update my PR in next couple days

MacArrow commented:
I've updated my patch but got stuck while testing it: with update to 5.4 I lost the ethernet network. I've updated my dts where I renamed ethernet to gmac0, tried various configurations of switch0 but no luck. The board I am using seems to be having mt7530 (at least openwrt build on 4.14 shows this switch when using swconfig) and port 0 is connected to the actual ethernet port, it has just one ethernet port, it is just a wireless access point. Who could help me with some points what to check for?

adrianschmutzler commented on target/linux/ramips/dts/mt7621_asus_rp-ac56.dts:
```diff
@@ -0,0 +1,244 @@
+                };
+        };
+};
+
```
Remove double empty line.

MacArrow replied:
Done

adrianschmutzler commented on target/linux/ramips/mt7621/base-files/etc/board.d/02_network:
```diff
@@ -46,6 +46,9 @@ ramips_setup_interfaces()
 		;;
+	asus,rp-ac56)
+		ucidef_set_interface_lan "lan"
+		;;
```
This can be merged with the block above (starting with mikrotik,routerboard-m11g|\). Due to alphabetic sorting, you then have to additionally move the entire block upwards.

MacArrow replied:
Done

adrianschmutzler commented:
> I've updated my patch but got stuck while testing it: with update to 5.4 I lost the ethernet network. > I've updated my dts where I renamed ethernet to gmac0, tried various configurations of switch0 but no luck. The board I am using seems to be having mt7530 (at least openwrt build on 4.14 shows this switch when using swconfig) and port 0 is connected to the actual ethernet port, it has just one ethernet port, it is just a wireless access point. Who could help me with some points what to check for?

Don't find this comment, though it reached me via e-mail. Did you delete it, or is this a GitHub issue?

Anyway, I do not see a problem from your configuration, so I assume it's hidden more deeply ...

adrianschmutzler commented on target/linux/ramips/image/rt3883.mk:
```diff
@@ -1,15 +1,15 @@
-define Build/mkrtn56uimg
-	$(STAGING_DIR_HOST)/bin/mkrtn56uimg $(1) $@
+define Build/asus-factory
+	$(STAGING_DIR_HOST)/bin/mkasusfw $(1) $(2) $@
```
I don't think $(2) will work, as I understand it, $(1) contains all argument.. Look here for example:
https://github.com/openwrt/openwrt/blob/master/target/linux/ramips/image/Makefile#L104

So, you won't need the $(2) at all if the $@ is not depending on it's position (haven't checked that).

MacArrow replied:
I've add a new arg for "asus-factory" tool so $(1) is a first arg, $(2) - second arg and $@ is the file name. Please correct me if I am wrong here...

adrianschmutzler replied:
But we are not using $(call ...) to call the function, so from my experience (though I do not find it in the docs right now) $(1) is the list of arguments, and you have to use
```
  $(eval firstarg=$(word 1,$(1)))
  $(eval secondarg=$(word 2,$(1)))
```
if you want the individual value. However, in your case this is obviously irrelevant and it would be fine to do just:
`$(STAGING_DIR_HOST)/bin/mkasusfw $(1) $@`

A nice example of what I mean is iodata-factory here: https://github.com/openwrt/openwrt/blob/master/target/linux/ramips/image/mt7621.mk#L51

Why is it still working for you? Because $(2) is just empty ...

MacArrow replied:
Thank you, got your point, fixed it

adrianschmutzler commented on target/linux/generic/files/drivers/mtd/mtdsplit/mtdsplit_uimage.c:
```diff
@@ -31,11 +31,20 @@
 #define IH_NMLEN		32	/* Image Name Length		*/
 
-#define IH_OS_LINUX		5	/* Linux	*/
+#define IH_OS_LINUX		5	/* Linux			*/
```
Any reason for that?

MacArrow replied:
The reason is consistency with lines on the top and below

adrianschmutzler replied:
Since this is not adjacent to any code you change, I'd just have kept it as it is.

But if you prefer it changed, I won't stop you.

MacArrow replied:
it was the only comment which did not follow the pattern on the top and the bottom so I decided to change it

adrianschmutzler commented on target/linux/generic/files/drivers/mtd/mtdsplit/mtdsplit_uimage.c:
```diff
@@ -55,6 +64,25 @@ struct uimage_header {
+        uint8_t         productid[IH_PRODLEN];
+        uint8_t         sub_fs;
+        uint32_t        ih_ksz;
+} asus_t;
```
Use tab indent as for the other struct/typedef

MacArrow replied:
done

adrianschmutzler commented on target/linux/ramips/dts/mt7621_asus_rp-ac56.dts:
```diff
@@ -0,0 +1,244 @@
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x4000000>;
+	};
```
memory node is not required (and thus generally omitted), as memory is auto-detected.

MacArrow replied:
Done

adrianschmutzler commented on target/linux/ramips/image/mt7621.mk:
```diff
@@ -9,6 +9,10 @@ DEFAULT_SOC := mt7621
 
+define Build/asus-factory
+        $(STAGING_DIR_HOST)/bin/mkasusfw $(1) $(2) $@
+endef
```
It's not necessary to keep this twice, just move it to image/Makefile when you touch it in the first patch of your series (and add a very short comment about that).

MacArrow replied:
Done, moved into image/Makefile, removed from other files

adrianschmutzler commented on target/linux/ramips/image/mt7621.mk:
```diff
@@ -145,6 +149,19 @@ define Device/asiarf_ap7621-nv1
+  DEVICE_MODEL := RP-AC56
+  IMAGE_SIZE := 16000k
+  BLOCKSIZE := 64k
+  DEVICE_PACKAGES := kmod-mt7603 kmod-mt76x2 wpad-basic kmod-i2c-core \
```
kmod-i2c-core will be automatically selected by kmod-i2c-ralink, kmod-sound-core will be automatically selected by kmod-sound-mt7620. So, it should be okay to remove both.

MacArrow replied:
Done

adrianschmutzler commented on target/linux/ramips/image/mt7621.mk:
```diff
@@ -145,6 +149,19 @@ define Device/asiarf_ap7621-nv1
+	kmod-i2c-ralink kmod-sound-mt7620 kmod-sound-core
+  IMAGES += factory.trx
+  IMAGE/factory.trx := append-kernel | append-rootfs | pad-rootfs | \
+	asus-factory -p RP-AC56 -f | check-size $$$$(IMAGE_SIZE)
```
check-size can be used without IMAGE_SIZE now.

MacArrow replied:
Done

adrianschmutzler commented on target/linux/ramips/mt7621/base-files/lib/preinit/04_handle_checksumming:
```diff
@@ -0,0 +1,27 @@
+		local rootfs_len=$(grep \"rootfs\" /proc/mtd | awk -F' ' '{print "0x"$2}')
+		local rootfs_data_len=$(grep \"rootfs_data\" /proc/mtd | awk -F' ' '{print "0x"$2}')
+		local offset=$(echo "$rootfs_len $rootfs_data_len 0x40" | awk -F' ' '{printf "%i",$1-$2-$3}')
+		local signature=$(dd if=$rootfs_mtd skip=$offset bs=1 count=4 2>/dev/null | hexdump -v -n 4 -e '1/1 "%02x"')
```
dd if=$rootfs_mtd skip=$offset bs=4 count=1 iflag=skip_bytes
Despite, can this be done with hexdump directly?

MacArrow replied:
Done, using hexdump directly, added that optimization, although I'd prefer readability vs performance here

adrianschmutzler commented on target/linux/ramips/mt7621/base-files/lib/preinit/04_handle_checksumming:
```diff
@@ -0,0 +1,27 @@
+		local offset=$(echo "$rootfs_len $rootfs_data_len 0x40" | awk -F' ' '{printf "%i",$1-$2-$3}')
+		local signature=$(dd if=$rootfs_mtd skip=$offset bs=1 count=4 2>/dev/null | hexdump -v -n 4 -e '1/1 "%02x"')
+		if [ "$signature" = "27051956" ]; then
+			dd conv=notrunc if=$firmware_mtd of=/tmp/header_orig.bin bs=1 count=64 2>/dev/null
```
bs=64 count=1

MacArrow replied:
Done although I'd prefer readability vs performance here

adrianschmutzler commented on target/linux/ramips/mt7621/base-files/lib/preinit/04_handle_checksumming:
```diff
@@ -0,0 +1,27 @@
+		local signature=$(dd if=$rootfs_mtd skip=$offset bs=1 count=4 2>/dev/null | hexdump -v -n 4 -e '1/1 "%02x"')
+		if [ "$signature" = "27051956" ]; then
+			dd conv=notrunc if=$firmware_mtd of=/tmp/header_orig.bin bs=1 count=64 2>/dev/null
+			dd conv=notrunc if=$rootfs_mtd skip=$offset of=/tmp/header_bkp.bin bs=1 count=64 2>/dev/null
```
bs=64 count=1 iflag=skip_bytes

MacArrow replied:
Done although I'd prefer readability vs performance here

adrianschmutzler replied:
I don't really see why it's harder to read now, because indeed readability is important for a code block like this. For the hexdump, I even think it's easier to read now.

Anyway, you need the iflag=skip_bytes, otherwise the offset will be the number of blocks, not bytes.

MacArrow replied:
yes, hexdump-based reading of the signature is much better and easier to read. what I meant is that if I use block size 64, it is not clear why block of 64, I have to use an additional flag and command line gets longer. And actually it was a good example: at the end I forgot to add a flag.

adrianschmutzler commented on target/linux/ramips/mt7621/base-files/lib/preinit/04_handle_checksumming:
```diff
@@ -0,0 +1,27 @@
+			dd conv=notrunc if=$rootfs_mtd skip=$offset of=/tmp/header_bkp.bin bs=1 count=64 2>/dev/null
+			if ! cmp -s /tmp/header_orig.bin /tmp/header_bkp.bin; then
+				echo "factory image for ASUS RP-AC56 detected, replacing uImage header..."
+				dd conv=notrunc if=$rootfs_mtd skip=$offset of=$firmware_mtd bs=1 count=64 2>/dev/null
```
See above.

MacArrow replied:
Same as above

MacArrow replied:
done

MacArrow commented:
BTW, it looks like an issue with jffs2 root_data partition is back (I believe a couple of months ago it was a patch reverted to address it), this build can not save configuration

adrianschmutzler commented:
> - no valid addresses found in 0x28, 0x2e, 0xe000 and 0xe006

This line can be removed from the commit message, as it result from the other ones.

Please remove or squash the merge commit in this PR.

adrianschmutzler commented on target/linux/ramips/dts/mt7621_asus_rp-ac56.dts:
```diff
@@ -0,0 +1,238 @@
+&pinctrl {
+	state_default: pinctrl0 {
+		gpio {
+			group = "uart2", "rgmii2";
```
group->groups

MacArrow replied:
Done

adrianschmutzler commented on target/linux/ramips/dts/mt7621_asus_rp-ac56.dts:
```diff
@@ -0,0 +1,238 @@
+};
+
+&pinctrl {
+	state_default: pinctrl0 {
```
Please use state_default directly, like in the other DTS files.

MacArrow replied:
Done

adrianschmutzler commented on target/linux/ramips/dts/mt7621_asus_rp-ac56.dts:
```diff
@@ -0,0 +1,238 @@
+
+	i2s_pins: i2s {
+		i2s {
+			group = "uart3";
```
group->groups

MacArrow replied:
Done

adrianschmutzler commented on tools/firmware-utils/Makefile:
```diff
@@ -42,6 +42,7 @@ define Host/Compile
 	$(call cc,lxlfw)
 	$(call cc,lzma2eva,-lz)
 	$(call cc,makeamitbin)
+	$(call cc,mkasusfw, -lz)
```
Remove space before "-lz"

MacArrow replied:
Done

adrianschmutzler commented on target/linux/ramips/dts/mt7621_asus_rp-ac56.dts:
```diff
@@ -0,0 +1,238 @@
+
+	chosen {
+		bootargs = "console=ttyS0,57600";
+	};
```
This has been moved to mt7621.dtsi in the meantime and can be dropped here.

MacArrow replied:
Done

adschm commented on target/linux/ramips/dts/mt7621_asus_rp-ac56.dts:
```diff
@@ -0,0 +1,234 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
```
dts-v1 is in mt7621 now, and can be removed here.

adschm commented on target/linux/ramips/dts/mt7621_asus_rp-ac56.dts:
```diff
@@ -0,0 +1,234 @@
+		compatible = "gpio-leds";
+
+		wlan2g_red {
+			label = "rp-ac56:red:wlan2g";
```
rp-ac56: prefix can be removed now for all leds, keeping just red:wlan2g etc.

adschm commented on target/linux/ramips/image/rt3883.mk:
```diff
@@ -1,15 +1,15 @@
-define Build/mkrtn56uimg
-	$(STAGING_DIR_HOST)/bin/mkrtn56uimg $(1) $@
+define Build/asus-factory
+	$(STAGING_DIR_HOST)/bin/mkasusfw $(1) $(2) $@
```
Still containing unavailable $(2) here (before moved to Makefile)
